docs/user/installation.rst

```diff
@@ -4,50 +4,239 @@
+<http://deeplearning.net/software/theano/install.html>`_ which cover installing
+the prerequisites for a range of operating systems, or ask for help on `our
+mailing list <https://groups.google.com/d/forum/lasagne-users>`_.
 
-Install from PyPI
-=================
 
-.. note:: Lasagne hasn't been released as of now.  The only way to
-          install it for now is from source.
+Prerequisites
+=============
 
-To install release 0.1 of Lasagne from PyPI, run the following
-commands:
+Python
+------
+
+Lasagne currently requires Python 2.7 or 3.4 to run. Please install Python via
+the package manager of your operating system if they are not included already.
+Use Python 2.7 if you're in doubt; it is still more widespread than Python 3.
+
+C compiler
+----------
+
+Theano requires a working C compiler, and numpy/scipy require a compiler as
+well if you install them via ``pip``. On Linux, the default compiler is usually
+``gcc``, and on Mac OS, it's ``clang``. Again, please install them via the
+package manager of your operating system.
+
+numpy/scipy + BLAS
+------------------
+
+Lasagne requires numpy of version 1.6.2 or above, and Theano also requires
+scipy 0.11 or above. Numpy/scipy rely on a BLAS library to provide fast linear
+algebra routines. They will work fine without one, but a lot slower, so it is
+worth getting this right (but less importantly so if you plan to use a GPU).
+
+If you install numpy and scipy via your operating system's package manager,
+they should link to the BLAS library installed in your system. If you install
+numpy and scipy via ``pip install numpy`` and ``pip install scipy``, make sure
+to have development headers for your BLAS library installed (e.g., the
+``libopenblas-dev`` package on Debian/Ubuntu) while running the installation
+command. Please refer to the `numpy/scipy build instructions
+<http://www.scipy.org/scipylib/building/index.html>`_ if in doubt.
+
+Theano
+------
+
+The version to install depends on the Lasagne version you choose, so this will
+be handled below.
+
+
+Stable Lasagne release
+======================
+
+.. note:: Lasagne hasn't been released yet. Please see the next section for
+    how to install the bleeding-edge version instead.
+
+Lasagne 0.1 requires a more recent version of Theano than the one available
+on PyPI. To install a version that is known to work, run the following command:
 
 .. code-block:: bash
 
   pip install -r https://raw.githubusercontent.com/Lasagne/Lasagne/0.1/requirements.txt
+
+A more recent version will often work as well, but at the time of writing, a
+simple ``pip install Theano`` will give you a version that is too old.
+
+To install release 0.1 of Lasagne from PyPI, run the following command:
+
+.. code-block:: bash
+
   pip install Lasagne==0.1
 
-Install from source
-===================
+If you do not have administrative rights on your machine, add ``--user`` to
+both commands to install into your home directory instead. To upgrade from an
```

Comment:
> I think a more correct cautious hint would be

That's a bit strong again for my taste :)

For me, everything's fine without virtualenv, mostly because I've never had a package complain about a dependency being too *new*. So for me, it works well to install the pip version of a module if the one from the Ubuntu repository is too old, and then it automatically takes precedence over the repository version so I don't have to uninstall the latter (possibly leaving Ubuntu package dependencies unresolved).
For some modules like numpy and scipy, I do site-wise installs because on our GPU machines, all users (yes, all users, I've asked them) want the latest version, can't install it with `--user` (since home directories are shared between multiple machines that sometimes have different BLAS libraries to link against) and would be very annoyed if they had to learn virtualenv and create a different environment per server.
So for me, the only use case for virtualenv is for checking whether a particular piece of software I wrote can be deployed on machines running an older version of Python, numpy or scipy. But this is not one of my day-to-day activities.

Probably you had a different experience, but I'd still try not to scare people too much ("dependency hell" and "fucking up your computer" sounds scary, even if worded differently), just give them the option of using virtualenv and let them decide. It should be as easy as possible to get from zero to Lasagne.

> http://www.dabapps.com/blog/introduction-to-pip-and-virtualenv-python/

Thanks, looks very good at first glance! I think I'll put this in the beginning.

> prefix the `--user` trick with "should you for some reason not be using virtualenv" or some such.

Good idea! I guess I'd still include something about administrative rights, but along with a warning that site-wide installs are seldom a good idea (this is something I can agree on, it doesn't make life any harder for newbies to recommend against site-wide installs).